kafka/kafka.go

```diff
@@ -168,3 +189,72 @@ func (t *topicSync) PublishBatch(ctx context.Context, batch []*bps.PubMessage) e
 	}
 	return t.producer.SendMessages(pmsgs)
 }
+
+// --------------------------------------------------------------------
+
+// Subscriber wraps a kafka consumer and implements the bps.Subscriber interface.
+// It starts consuming from the newest offset (so from message, received after connecting to kafka).
+type Subscriber struct {
+	consumer      sarama.Consumer
+	initialOffset int64
+}
+
+// NewSubscriber inits a new subscriber.
+func NewSubscriber(addrs []string, config *sarama.Config) (*Subscriber, error) {
+	consumer, err := sarama.NewConsumer(addrs, config)
+	if err != nil {
+		return nil, err
+	}
+	return &Subscriber{
+		consumer:      consumer,
+		initialOffset: config.Consumer.Offsets.Initial,
+	}, nil
+}
+
+// Subscribe implements the bps.Subscriber interface.
+func (s *Subscriber) Subscribe(ctx context.Context, topic string, handler bps.Handler) error {
+	parts, err := s.consumer.Partitions(topic)
```

Comment:
so one thing that you will need to do as well is create an `OffsetManager` (sounds like we need a separate `Client` as well to be shared between `Consumer` and OM)

Reply:
Agreed not to do for initial merge/release ^^